### PR TITLE
Fix: Remove unused form error-handling; add new placeholder + regex

### LINKED ISSUE #1007: Remove unused form with-errors logic/code
With Courtesy Cards, we have the browser-based form checks for required fields and `sub` pattern (regex) -- there isn't a way for the user to submit the form with bad inputs, assuming the browser is working as advertised.

Further, when the data gets to our Server, it will be hashed -- there isn't a way for the Server to know if the format of user data was incorrect and/or missing.

All that is to say - we have a ton of HTML/CSS/JS/Python for dealing with form errors that really isn't needed anymore. Let's clean this up.

## Acceptance Criteria

- [x] There is not HTML/CSS/JS related to form errors
- [x] The Eligibility form does not deal with form/API errors

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #1007 

## What this PR does
- Remove form `with-errors` related HTML, CSS, JS
- Add sample placeholder and regex

## How to test
- Go through both forms and ensure no regressions in style and function
- Go through Input form: Test an input that fails the regex

image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@thekaveman @angela-tran small mostly deletion pr ready for review. It would be helpful for me to have this in dev before I work on forms PR.

**Review by angela-tran — COMMENTED** [review]
I think we should also delete the `add_api_errors` method in `EligibilityVerificationForm`

File: benefits/core/templates/core/includes/form.html [hunk 1]
```diff
@@ -7,70 +7,57 @@
...
-    </form>
-
-  {% endif %}
+    {% endif %}
+
+    <script>
+          $(function(){
+              $("form[action='{{form_action}}']").on("submit", function(e) {
+                  if ("{{ form.submitting_value }}" !== "") {
+                      $("button[data-action='submit']", this)
```
Comment (thekaveman) [inline benefits/core/templates/core/includes/form.html:52]: I don't think this is a regression from this PR, but I just noticed... not all of the `button`s on this template have the `data-action` attribute, so this selector would not find it sometimes

**Review by thekaveman — CHANGES_REQUESTED** [review]
No change required for this, but a comment on commit 4413ea0ec6f4e40f5834450cfcfb6d917c7d5bc3 - the message doesn't seem clear/accurate, and it was a huge amount of changes... maybe for next time, break something like that up into a few smaller chunks: dealing with the form separately from the JS for example, and the whitespace/formatting separately from both.

One requested change below, otherwise this looks good!

File: benefits/core/templates/core/includes/form.html (hunk 1 above)
Comment (machikoyasuda) [reply benefits/core/templates/core/includes/form.html:52]: yea sorry the rebase commits https://github.com/cal-itp/benefits/commit/4413ea0ec6f4e40f5834450cfcfb6d917c7d5bc3 are all stemming from difficulty of managing several open PRs + rebasing dev + slow GitHub Actions. and yes, the spacing on this form template is really confusing.

Comment (machikoyasuda) [reply benefits/core/templates/core/includes/form.html:52]: I can take care of and test this Form button issue in either https://github.com/cal-itp/benefits/issues/947 and https://github.com/cal-itp/benefits/issues/972. The Button CSS code, Button viewmodel and Form HTML code have not been changed yet in this Sprint and this is the first of many PRs to start cleaning this all up.

Comment (angela-tran) [reply benefits/core/templates/core/includes/form.html:52]: Is this actually the cause of https://github.com/cal-itp/benefits/issues/948?

Comment (machikoyasuda) [reply benefits/core/templates/core/includes/form.html:52]: @angela-tran probably

**Pull request closed without merging**